src/GraphQL.Extensions.DI.Microsoft/ScopedFieldResolver.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using GraphQL.Resolvers;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.Extensions.DI.Microsoft
+{
+    public class ScopedFieldResolver<TReturnType> : FuncFieldResolver<TReturnType>
```

Comment:
Maybe it is worth making `ScopedFieldResolver`s `internal` ?

Reply:
The same for both `ScopedAsyncFieldResolver`s.

Reply:
I would definitely use them in my implementation.

Reply:
How? As base classes?

Reply:
Here is one example:

https://github.com/Shane32/GraphQL.DI/blob/1779914c4ebc52305df26497dec0ffc05cbfa391/src/DIObjectGraphType.cs#L228